proxy/src/serverless/backend.rs

```diff
@@ -33,21 +36,22 @@ pub struct PoolingBackend {
 }
 
 impl PoolingBackend {
-    pub async fn authenticate(
+    pub async fn authenticate_with_password(
         &self,
         ctx: &RequestMonitoring,
         config: &AuthenticationConfig,
-        conn_info: &ConnInfo,
+        user_info: &ComputeUserInfo,
+        password: &[u8],
```

Comment:
We should consider introducing some type that protects passwords, like `secrecy::Secret` or a newtype around it.